src/mappings/Rewards.ts

```diff
@@ -1,55 +1,72 @@
 import {HistoryElement} from '../types';
 import {SubstrateEvent} from "@subql/types";
-import {callsFromBatch, eventId, isBatch, distinct, timestamp} from "./common";
+import {callsFromBatch, eventIdFromBlockAndIdx, isBatch, distinct, timestamp, eventId} from "./common";
 import {CallBase} from "@polkadot/types/types/calls";
 import {AnyTuple} from "@polkadot/types/types/codec";
 
 function isPayoutStakers(call: CallBase<AnyTuple>): boolean {
     return call.method == "payoutStakers"
 }
 
-export async function handleReward(event: SubstrateEvent): Promise<void> {
-    const {event: {data: [account, newReward]}} = event;
+export async function handleReward(rewardEvent: SubstrateEvent): Promise<void> {
+    const {event: {data: [account, newReward]}} = rewardEvent;
+    let blockNumber = rewardEvent.block.block.header.number.toString()
+    let blockTimestamp = timestamp(rewardEvent.block)
 
-    const element = new HistoryElement(eventId(event));
+    let element = await HistoryElement.get(eventId(rewardEvent))
 
-    element.address = account.toString()
-    element.timestamp = timestamp(event.block)
+    if (element != undefined) {
+        // already processed reward previously
+        return;
+    }
 
-    const cause = event.extrinsic
+    const cause = rewardEvent.extrinsic
     const causeCall = cause.extrinsic.method
 
-    let validator;
+    let payoutedValidators: string[]
 
     if (isPayoutStakers(causeCall)) {
         const [validatorAddress,] = cause.extrinsic.method.args
```

Comment:
Wow, are we really getting address here? I think it should be accountId

Reply:
It seems to be converted address after you do .toString()

Reply:
Great!